Firestore/core/src/remote/remote_event.cc

```diff
@@ -254,22 +255,32 @@ void WatchChangeAggregator::HandleExistenceFilter(
 
 bool WatchChangeAggregator::ApplyBloomFilter(
     const ExistenceFilterWatchChange& existence_filter, int current_count) {
-  const absl::optional<BloomFilter>& bloom_filter =
-      existence_filter.filter().bloom_filter();
-  if (!bloom_filter.has_value()) {
+  const absl::optional<BloomFilterParameter>& bloom_filter_parameter =
+      existence_filter.filter().bloom_filter_parameter();
+  if (!bloom_filter_parameter.has_value()) {
+    return false;
+  }
+
+  util::StatusOr<BloomFilter> maybe_bloom_filter =
+      BloomFilter::Create(bloom_filter_parameter.value().bitmap,
+                          bloom_filter_parameter.value().padding,
+                          bloom_filter_parameter.value().hash_count);
+  if (!maybe_bloom_filter.ok()) {
+    LOG_WARN("Creating BloomFilter failed: %s",
+             maybe_bloom_filter.status().error_message());
     return false;
   }
 
   int removed_document_count = FilterRemovedDocuments(
-      bloom_filter.value(), existence_filter.target_id());
+      std::move(maybe_bloom_filter).ValueOrDie(), existence_filter.target_id());
```

Comment:
Remove the `std::move()` here because `FilterRemovedDocuments()` takes a const reference argument.